test/unit/utils/dom-data.test.js

```diff
@@ -21,3 +23,76 @@ QUnit.test('should get and remove data from an element', function(assert) {
 
   assert.notOk(DomData.hasData(el), 'cached item emptied');
 });
+
+QUnit.done(function(details) {
```

Comment:
It doesn't seem like tests are supposed to be run in `QUnit.done` and it makes re-running them a bit awkward, but I don't see another way to check for memory leaks other than waiting until done.